src/screens/CardScreen.tsx

```diff
@@ -0,0 +1,339 @@
+      return;
+    }
+    let item = benefitsToAdd.find((x) => x.id === benefit.publicId);
+    item ? item.amount++ : benefitsToAdd.push({ id: benefit.publicId, amount: 1 });
+    setBalance(accountBalance - benefit.price);
+    console.log(benefitsToAdd);
+  };
+
+  const OnSave = () => {
+    benefitsToAdd.forEach((benefit) => {
+      let obj = Card.content.inventory.find((x) => x.id === benefit.id);
+      obj ? (obj.amount += benefit.amount) : Card.content.inventory.push(benefit);
+    });
+    Card.content.points = accountBalance;
+    benefitsToAdd = [];
+    //console.log(Card.content.inventory);
+  };
+
+  const OnCancel = () => {
+    benefitsToAdd = [];
+    setBalance(Card.content.points);
+  };
+
+  return Card.type === 'virtual' ? (
+    <View style={StyleBase.container}>
+      <StatusBar barStyle='default' />
+      {screenState === 'card' && (
+        <>
+          <TopBar iconLeft={'arrow-left'} onPressLeft={() => onPressBack(navigation)} />
+          {/* todo: image as Card.businessDetails.iconImageId */}
+          <CardTile
+            containerStyle={[styles.cardTile, !Card.isAdded && { paddingBottom: 75 }]}
+            image={Card.content.businessDetails.bannerImageId}
+            tileStyle={{ width: '88.66%' }}
+          />
+          {Card.isAdded && (
+            <Tile style={styles.cardInfo}>
+              <View style={styles.accountTileContainer}>
+                <Text style={styles.text}>Account balance</Text>
+                <View style={{ width: '35%', alignItems: 'flex-end' }}>
+                  <Text style={styles.text}>{accountBalance}</Text>
+                </View>
+              </View>
+            </Tile>
+          )}
+          <View style={styles.buttonsContainer}>
+            <Pressable
+              onPress={() => {
+                setInfo('business');
+              }}
+            >
+              <Tile
+                style={[
+                  styles.button,
+                  cardInfoState === 'benefits' && { backgroundColor: colors.swPaleViolet },
+                ]}
+              >
+                <Text style={styles.text}>Business</Text>
+              </Tile>
+            </Pressable>
+            <Pressable
+              onPress={() => {
+                setInfo('benefits');
+              }}
+            >
+              <Tile
+                style={[
+                  styles.button,
+                  cardInfoState === 'business' && { backgroundColor: colors.swPaleViolet },
+                ]}
+              >
+                <Text style={styles.text}>Benefits</Text>
+              </Tile>
+            </Pressable>
+          </View>
+          <View style={[styles.container, !Card.isAdded && { height: '50%' }]}>
+            {cardInfoState === 'business' && (
+              <>
+                <BoxContainer style={styles.boxContainer}>
+                  <Text style={[styles.text, { paddingBottom: 40 }]}>
+                    {Card.content.businessDetails.name}
```

Comment:
W wielu miejscach pojawia się takie mocne odwołanie do `Card.content.kolejnepole.kolejnepole`.

Możesz to zdestrukturyzować przed returnem w tym komponencie.

np: 

`const {content : {businessDetails, inventory } } = Card`

Reply:
I wtedy odwoływać się do `businessDetails i inventory` elegancko.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
Zrobiłbym tak z każdym polem `Card`, który tutaj wykorzystujemy.